samples/pubsub/workerpool/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"gocloud.dev/internal/workerpool"
```

Comment:
Since workerpool is internal, this example can't be used outside of the repo.

I thought you were going to use workerpool to build something pubsub-specific, either in the `pubsub` package or a subpackage (`pubsub/helpers`?). Something like the callback version of Receive that you originally wanted to do.  

I think I prefer that to making workerpool public, but I could be convinced. One concern I have is that workerpool is a simple, elegant but very general piece of functionality that we'd be committing to supporting if it moved out of internal. @zombiezen, what do you think? 

Honestly, I think my favorite solution is just "inlining" workerpool.Run here. That demonstrates the bcmills pattern clearly, with no abstraction layer, and doesn't commit us to maintaining a general-purpose package or adding any API to `pubsub`.

Reply:
Good point. I'll try adding `pubsub.RunWorkerPool(sub, handleMessage)` and we can see how it looks.

Reply:
Done. PTAL.

Reply:
I disagree about the utility of inlining the worker pool code. Concurrency logic is notoriously difficult to get right and since we've managed to abstract it in this case it's a clear win to have it as a function.

Reply:
The argument about a maintenance burden seems unconvincing since there is no clear vector of bit rot for a function like `workerpool.Run` with no dependencies on anything other than `context`.

Reply:
I'm more in favor of inlining the `workerpool` run for sample code, as then it's more clear what is going on. I don't see "concurrency being difficult" as being a reason to not show what's going on for illustration, especially if the pattern is well-known.

That said, one bit of management I really would like to see is coming up with a better way of doing [what I did for Contribute Bot](https://github.com/google/go-cloud/blob/master/internal/contributebot/main.go#L63-L65) where I want a process to subscribe on a topic and also serve HTTP health checks for Kubernetes (think `server` package for subscriber services). Just having the worker pool isn't really enough to justify a library function in the Go CDK.

Reply:
We've moved on from the inline/noinline discussion to making a helper function.

I don't see why we need to provide a way to interleave receiving messages from a subscription with other activities like health checks. That is one of the nice things about concurrency: we can interleave any activities we want by running each in its own goroutine. That is what you do in the linked contributebot code, and I don't know what a better way of doing that would be.

Reply:
The code that I linked is subtly wrong in that it does not synchronize or shutdown the health check listening port. I would expected almost all pub/sub subscriber processes running in an orchestrated environment would need to write similar code.

That said, I don't have enough insight here to give any compelling advice, just lots of unorganized input. Consider my two cents added. :)

Reply:
At a glance I figure an errgroup would do the trick.